Sources/StreamChatUI/Utils/Debouncer.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+public protocol Debouncing {
+    // Define a type alias for the handler closure.
+    typealias Handler = () -> Void
+
+    // Method to debounce the handler closure.
+    func debounce(
+        _ handler: @escaping Handler
+    )
+
+    // Method to cancel any pending debounce calls.
+    func cancel()
+}
+
+open class Debouncer: Debouncing {
```

Comment:
This one should not be `open` for sure, a debouncer is a debouncer and should not have any customizability 👍